WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelector.swift

```diff
@@ -14,6 +14,10 @@ struct ProductSelector: View {
     ///
     @State var isShowingVariationList: Bool = false
 
+    /// View model to use for the variation list, when it is shown.
+    ///
+    @State var variationListViewModel: ProductVariationSelectorViewModel?
```

Comment:
Do you think we can provide a custom binding of `isShowingVariationList` in order to nil the `variationListViewModel` once it has been dismissed / closed / navigated away?

Reply:
I thought about this a bit and felt divided about it:

On one hand, it isn't strictly necessary to nil `variationListViewModel` since it is updated with the newly selected view model any time `isShowingVariationList` is set to `true`. (Any previous value is ignored until it's updated again when a new variable product is selected in the list.)

On the other hand, it seems reasonable to make the connection between these properties more explicit. Rather than a custom binding, if we want to do this I think we could use `.onChange(of:perform:)` to nil `variationListViewModel` as an explicit side effect of `isShowingVariationList` being changed.

However, after trying that out I don't think it's worth doing this. If we nil `variationListViewModel` every time we navigate back, the `LazyNavigationLink` isn't kept in the view hierarchy (because of the nil check `if let variationListViewModel = variationListViewModel`). If it isn't already in the view hierarchy, there isn't any animation when going from the product list to the variation list. The solution for that would be to accept an optional view model in `ProductVariationSelector` and I think that creates unnecessary complexity there.

What do you think? It's possible to do, but I'm not sure it's worth it in this case.

Reply:
Well, if It's causing animation problems maybe it's ok to leave it as it is?
I'm not sure I totally understand why the navigation link has to exists conditionally tho 😅